### PR TITLE
Ut coverage in edge/pkg/edgehyb/common/http/

### LINKED ISSUE #2245: edge/pkg/edgehub: improve test coverage
Add more tests to import test coverage. Details refer to #2110 .

/help

### PR DESCRIPTION
**Pull request by jmguzik**
**What type of PR is this?**
/kind test

**What this PR does / why we need it**:
Ut coverage improved in edge/pkg/edgehyb/common/http/ (from ~30% to 97%)

**Which issue(s) this PR fixes**:

Fixes #2245

**Special notes for your reviewer**:
Difficult to achieve 100% since it would require real HTTP interaction.

**Does this PR introduce a user-facing change?**:
no

```release-note
NONE
```

### CODE REVIEW

**Comment by jmguzik** [comment]
Informing @daixiang0 as an original author of issue #2245

File: edge/pkg/edgehub/common/http/http_test.go [hunk 1]
```diff
@@ -105,3 +110,168 @@ func TestNewHTTPSClient(t *testing.T) {
 		})
 	}
 }
+
+//TestNewHTTPClientWithCA() tests the creation of a new HTTP using filled capem
```
Comment (fisherxu) [inline edge/pkg/edgehub/common/http/http_test.go:114]:
```suggestion
// TestNewHTTPClientWithCA() tests the creation of a new HTTP using filled capem
```

File: edge/pkg/edgehub/common/http/http_test.go [hunk 2]
```diff
@@ -105,3 +110,168 @@ func TestNewHTTPSClient(t *testing.T) {
...
+func TestNewHTTPClientWithCA(t *testing.T) {
+	err := util.GenerateTestCertificate("/tmp/kubeedge/testData/", "edge", "edge")
```
Comment (fisherxu) [inline edge/pkg/edgehub/common/http/http_test.go:116]: Can we use a const here?

File: edge/pkg/edgehub/common/http/http_test.go [hunk 3]
```diff
@@ -105,3 +110,168 @@ func TestNewHTTPSClient(t *testing.T) {
...
+	req, err := http.NewRequest(Method, URL, bytes.NewReader([]byte{}))
+	if err != nil {
+		t.Errorf("Error in creating new http request message: %v", err)
+		return
+	}
+
+	resp, respErr := SendRequest(req, httpClient)
+	if resp != nil && respErr == nil {
+		t.Errorf("Error, response should not come as data is not valid")
+		return
```
Comment (fisherxu) [inline edge/pkg/edgehub/common/http/http_test.go:275]: Remove the return here? :)

File: edge/pkg/edgehub/common/http/http_test.go [hunk 4]
```diff
@@ -105,3 +110,168 @@ func TestNewHTTPSClient(t *testing.T) {
...
+				t.Errorf("BuildRequest() Header = %v, want %v", got, tt.want.Header)
+			}
+			if !reflect.DeepEqual(got.Body, tt.want.Body) {
+				t.Errorf("BuildRequest() Body = %v, want %v", got, tt.want.Body)
+			}
+		})
+	}
+}
+
+//TestSendRequestFailure() uses fake data and expects function to fail
```
Comment (fisherxu) [inline edge/pkg/edgehub/common/http/http_test.go:259]:
```suggestion
// TestSendRequestFailure() uses fake data and expects function to fail
```

File: edge/pkg/edgehub/common/http/http_test.go [hunk 5]
```diff
@@ -105,3 +110,168 @@ func TestNewHTTPSClient(t *testing.T) {
...
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("NewHTTPClientWithCA() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+//TestBuildRequest() tests the process of message building
```
Comment (fisherxu) [inline edge/pkg/edgehub/common/http/http_test.go:186]:
```suggestion
// TestBuildRequest() tests the process of message building
```

**Review by fisherxu — COMMENTED** [review]
Thanks @jmguzik

File: edge/pkg/edgehub/common/http/http_test.go (hunk 5 above)
Comment (jmguzik) [reply edge/pkg/edgehub/common/http/http_test.go]: Thanks for the review. I hope all your comments are included in my new commit :)

**Pull request merged**